dpbench/infrastructure/benchmark_validator.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-FileCopyrightText: 2022 - 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Validation tools for comparing arrays."""
+
+import logging
+from numbers import Number
+from typing import Union
+
+import numpy as np
+
+
+def validate_results(
+    expected: dict[str, any], actual: dict[str, any], rel_error=1e-05
+) -> bool:
+    """Checks if expected equals actual with certain precision.
+
+    Args:
+        expected: expected values.
+        actual: actual values.
+        rel_error: maximum acceptable relative error.
+
+    Returns: true, if provided data is equal.
+    """
+    if not expected:
+        return False
+
+    try:
+        for key in expected.keys():
+            valid = validate_two_lists_of_array(
+                expected[key], actual[key], rel_error=rel_error
+            )
+            if not valid:
+                logging.error(
+                    (
+                        "Output did not match for {0}. "
+                        + "Expected: {1} Actual: {2}"
+                    ).format(key, expected[key], actual[key])
+                )
+        return valid
+    except Exception as e:
+        logging.error(f"Exception during validation {e.args}")
+        return False
+
+
+def validate_two_lists_of_array(
+    expected: Union[list[np.ndarray], np.ndarray],
+    actual: Union[list[np.ndarray], np.ndarray],
+    rel_error,
+) -> bool:
+    """Checks if expected equals actual with certain precision.
+
+    Compares two arrays or two lists of arrays and validates if
+    the arrays in each list have data that are either the same or close
+    enough.
+
+    Args:
+        expected: list of arrays with the reference results of a specific
+            benchmark
+        actual: list of arrays array with the results generated by the
+            framework's implementation of the benchmark
+        rel_error: maximum acceptable relative error.
+
+    Returns: true, if provided data is equal.
+    """
+    if not isinstance(expected, (tuple, list)):
+        expected = [expected]
+    if not isinstance(actual, (tuple, list)):
+        actual = [actual]
+    valid = True
+    for r, v in zip(expected, actual):
+        if not np.allclose(r, v):
+            try:
+                import cupy
+
```

Comment:
Why `import` cupy? I recall how it was done originally in npbench and at some point I had changed it. The asnumpy should happen at the call site and the validation function should always expect NumPy arrays. 

How does it work for dpnp?

Reply:
Aha I checked the old `utilities` module and the same code existed there too. So maybe I never made the change. The question remains valid though, we should not have the `cupy` specific code here.

Reply:
Yeah, I agree, it's just was copy-paste)